pkg/controller/perconaservermongodb/version.go

```diff
@@ -41,15 +39,25 @@ func (r *ReconcilePerconaServerMongoDB) sheduleEnsureVersion(cr *api.PerconaServ
 
 	if ok {
 		log.Info(fmt.Sprintf("remove job %s because of new %s", schedule.CronShedule, cr.Spec.UpgradeOptions.Schedule))
-		r.deleteEnsureVersion(schedule.ID)
+		r.deleteEnsureVersion(cr, schedule.ID)
 	}
 
-	log.Info(fmt.Sprintf("add new job: %s", cr.Spec.UpgradeOptions.Schedule))
 	id, err := r.crons.crons.AddFunc(cr.Spec.UpgradeOptions.Schedule, func() {
-		r.statusMutex.Lock()
-		defer r.statusMutex.Unlock()
+		nn := types.NamespacedName{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		}
 
-		if !atomic.CompareAndSwapInt32(&r.updateSync, updateDone, updateWait) {
+		l, ok := r.lockers.Load(nn.String())
```

Comment:
Should we create an object just for the concatenation of the strings?
Maybe:
```go
nn := cr.Name + "/" + cr.Namespace
l, ok := r.lockers.Load(nn)
...
```

Reply:
Just want to be consistent. I have same on controller.
And do not want somebody accidentally change it to something else. Using `types.NamespacedName` make it more solid (maybe if somebody want to change he will think one more time why `types.NamespacedName` used here).

Reply:
yes, make sense